client/alloc_runner.go

```diff
@@ -257,6 +269,7 @@ func (r *AllocRunner) setTaskStatus(taskName, status, desc string) {
 		Description: desc,
 	}
 	r.taskStatusLock.Unlock()
+	r.SaveState(taskName)
```

Comment:
@dadgar We are also logging and adding the error to multierror, and it was being repeated in two places so it made sense to extract that as a method.